src/middleware/socketAuth.js

```diff
@@ -0,0 +1,33 @@
+const socketio = require('feathers-socketio');
+
+module.exports = function () {
+	const app = this;
+	app.configure(socketio((io) => {
+		io.use(function (socket, next) {
+			let jwt = extractTokenFromCookies(socket.handshake.headers.cookie);
+			app.passport.authenticate("jwt")({headers: {authorization: jwt}})
+				.then((payload) => {
+					socket.client.userId = payload.data.account.userId;
+					next();
+				})
+				.catch((error) => {
+					next(new Error('Authentication error'));
+				});
+		});
+    }));
+
+	function extractTokenFromCookies(cookies) {
+		try {
+			cookies = cookies.split(';');
+			let jwt = undefined;
+			cookies.map(cookie => {
+				if (cookie.includes('jwt')) {
```

Comment:
What if some other cookie part, like a big hash, contains "jwt" before the actual jwt? Might be rarely problematic.